memgpt/server/rest_api/presets/index.py

```diff
@@ -0,0 +1,121 @@
+import uuid
+from functools import partial
+from typing import List, Optional, Dict, Union
+
+from fastapi import APIRouter, Body, Depends, Query, HTTPException, status
+from fastapi.responses import JSONResponse
+from pydantic import BaseModel, Field
+
+from memgpt.data_types import Preset  # TODO remove
+from memgpt.models.pydantic_models import PresetModel
+from memgpt.server.rest_api.auth_token import get_current_user
+from memgpt.server.rest_api.interface import QueuingInterface
+from memgpt.server.server import SyncServer
+from memgpt.constants import DEFAULT_HUMAN, DEFAULT_PERSONA
+from memgpt.utils import get_human_text, get_persona_text
+
+router = APIRouter()
+
+"""
+Implement the following functions:
+* List all available presets 
+* Create a new preset 
+* Delete a preset 
+* TODO update a preset 
+"""
+
+
+class ListPresetsResponse(BaseModel):
+    presets: List[PresetModel] = Field(..., description="List of available presets.")
+
+
+class CreatePresetsRequest(BaseModel):
+    # TODO is there a cleaner way to create the request from the PresetModel (need to drop fields though)?
+    name: str = Field(..., description="The name of the preset.")
+    id: Optional[Union[uuid.UUID, str]] = Field(default_factory=uuid.uuid4, description="The unique identifier of the preset.")
```

Comment:
@sarahwooders `test_client` was failing until I added the ability to pass `str` as a type. The failure was happening inside the REST client method that creates this `CreatePresetsRequest` object to prepare the payload, then dumps it to `json` for the POST request (the error specifically was a "cannot serialize UUID type to JSON).